LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/RadioMetadata.cs

```diff
@@ -3,35 +3,7 @@
 
 namespace LoRaWan.NetworkServer.BasicsStation
 {
-    public class RadioMetadata
-    {
-        public DataRateIndex DataRate { get; }
-        public Hertz Frequency { get; }
-        public RadioMetadataUpInfo UpInfo { get; }
+    public record RadioMetadata(DataRateIndex DataRate, Hertz Frequency, RadioMetadataUpInfo UpInfo);
```

Comment:
I would recommend making this a regular record as opposed to a positional record (the latter is good when there's a natural order to fields, but I don't think we can assume that here):

```suggestion
    public sealed record RadioMetadata
    {
        public RadioMetadata(DataRateIndex dataRate, Hertz frequency, RadioMetadataUpInfo upInfo)
        {
            DataRate = dataRate;
            Frequency = frequency;
            UpInfo = upInfo;
        }

        public DataRateIndex DataRate { get; init; }
        public Hertz Frequency { get; init; }
        public RadioMetadataUpInfo UpInfo { get; init; }
    }
```